File: skills/logging-review/samples/dotnet/conformant/OrderProcessor.cs
using Microsoft.Extensions.Logging;

namespace LoggingSamples.Conformant;

public sealed record Order(string Id, string CustomerEmail);

public sealed partial class OrderProcessor
{
    private readonly ILogger<OrderProcessor> _logger;

    public OrderProcessor(ILogger<OrderProcessor> logger) => _logger = logger;

    [LoggerMessage(EventId = 1001, EventName = "OrderConfirmed",
        Level = LogLevel.Information, Message = "Order confirmed {OrderId}")]
    private partial void LogOrderConfirmed(string orderId);

    // Auditable event -> INFO, no PII: OrderId only, never CustomerEmail.
    public void ConfirmOrder(Order order) => LogOrderConfirmed(order.Id);
}

File: skills/logging-review/samples/react-ts/conformant/orderProcessor.ts
export interface Logger {
  info(event: string, fields: Record<string, unknown>): void;
  warn(event: string, fields: Record<string, unknown>): void;
  error(event: string, fields: Record<string, unknown>): void;
}

export class OrderProcessor {
  constructor(private readonly logger: Logger) {}

  // Auditable event -> info, no PII: orderId only, never customerEmail.
  confirmOrder(order: { id: string; customerEmail: string }): void {
    this.logger.info("OrderConfirmed", { orderId: order.id });
  }
}
